src/components/Widgets/ListControl.js

```diff
@@ -102,8 +102,8 @@ export default class ListControl extends Component {
     return (e) => {
       e.preventDefault();
       const { value, metadata, onChange, forID } = this.props;
-      const parsedMetadata = { [forID]: metadata.removeIn(value.get(index).valueSeq()) };
-      onChange(value.remove(index), parsedMetadata);
+      const parsedMetadata = metadata && { [forID]: metadata.removeIn(value.get(index).valueSeq()) };
+      onChange(value.remove(index), parsedMetadata || undefined);
```

Comment:
is that `|| undefined` really necessary? `parsedMetadata` will be null if there is no metadata, the onChange callback should know how to handle that, shouldn't it?